Source/JavaScriptCore/runtime/JSArray.cpp

```diff
@@ -566,16 +565,17 @@ JSArray* JSArray::fastToReversed(JSGlobalObject* globalObject, uint64_t length)
         butterfly->setVectorLength(vectorLength);
         butterfly->setPublicLength(length);
 
-        auto resultData = butterfly->contiguous().data();
-        memcpy(resultData, srcData, sizeof(JSValue) * length);
-        Butterfly::clearOptimalVectorLengthGap(type, butterfly, vectorLength, length);
-
         if (hasDouble(indexingType)) {
+            ASSERT(!containsHole(this->butterfly()->contiguousDouble().data(), static_cast<uint32_t>(length)));
             auto data = butterfly->contiguousDouble().data();
+            memcpy(data, this->butterfly()->contiguousDouble().data(), sizeof(double) * length);
             std::reverse(data, data + length);
-        } else
+        } else {
+            auto resultData = butterfly->contiguous().data();
+            copyArrayElements<ArrayFillMode::Undefined, NeedsGCSafeOps::No>(resultData, 0, this->butterfly()->contiguous().data(), 0, length, type);
```

Comment:
Filling element with undefined is not correct when the result array is Int32Shape.

Reply:
Thank you!  I update it to if result is an int32 and a hole is found, that the resulting array is converted to a contiguous shape. What do you think?

Reply:
And added some tests